document/CoreFormat.mdk

```diff
@@ -303,51 +316,122 @@ an analysis run. The value of `invocationInfo` would have to be combined with in
+but it may contain properties representing all scanned files (whether or not results were detected in those files),
+or for any subset of those files.
+
+~ IsoNumberedNote { @IsoNumberedNote=1}
+`fileReference` objects contain information that is useful for viewers.
+Viewers will be able to provide the most information to users if the `fileInfo`
+property is present and contains information for every file in which results were detected.
 ~
 
-The `analysisTargets` array shall be empty if no analysis targets were scanned in the course of the run.
+Each property name in the `fileInfo` object shall be the URI of a scanned file.
+No two of these property names shall be equivalent as defined in &sect;[#uriProperties].
 
-~ IsoNumberedNote
-This could happen if the command line specified a wildcard such as *.cc for the input files,
-and no files matched the wildcard.
-~
+Each property value in the `fileInfo` object shall be an array of `fileReference` objects (&sect;[#fileReference])
+which contain information about the file identified by the URI in the property name.
 
-~ IsoExample
+If the scanned file is not nested within another file, then the array shall consist of a single
+element, and the property name shall be the URI of the scanned file.
+If the absolute location of the scanned file is available, the URI should be an absolute URI;
+otherwise, the URI shall be a relative URI.
+
+~ IsoNumberedExample { @IsoNumberedExample=1}
 
 ```
-[
-    {
-        "uri": "file:///C:/Code/a.cc",
-        "hashes": [
+runInfo: {
+    fileInfo: {
+        "file:///C:/Code/main.cc": [
             {
-                "value": "b13ce2678a8807ba0765ab94a0ecd394f869bc81",
-                "algorithm": "SHA-256"
+                "mimeType": "text/x-c",
+                "hashes": [
+                    {
+                        "value": "b13ce2678a8807ba0765ab94a0ecd394f869bc81",
+                        "algorithm": "SHA-256"
+                    }
+                ]
             }
         ]
-    },
+    }
+}
+```
+~
 
-    {
-        "uri": "file:///C:/Code/b.cc",
-        "hashes": [
+If the scanned file is nested within another file (for example, within a compressed container),
+then the first `fileReference` object in the array shall refer to the outermost container,
+the next one shall refer to a file nested immediately within that container, and so on,
+with the last element of the array referring to the most deeply nested file.
+
+In this case, the property name shall be the absolute URI of the outermost containing
+file, together with a fragment that describes the nesting of the scanned file within
+its parent or parents.
+If the scanned file is nested more than one level deep in the outermost containing file,
+the fragments representing each level of nesting may be combined in any way desired,
+as long as no two of the resulting property names are equivalent as defined in &sect;[#uriProperties].
```

Comment:
this doesn't seem clear. i think you're saying that each level of nesting may be combined in any way, as long as the net result does not create a complete fragment that, in combination with an identical outmost container URI, results in an identical string (which would cause collisions in the dictionary. If this is, in fact, what you mean, it's not clear here (but may be in #uriProperties, which i will look at now.